node/src/signature_mngr.rs

```diff
@@ -109,8 +121,8 @@ impl Actor for SignatureManager {
             })
             .map_err(|e| log::error!("Couldn't initialize Signature Manager: {}", e))
             .into_actor(self)
-            .map(|secret_key, _, ctx| {
-                ctx.notify(SetKey(secret_key));
+            .map(|secret_key, act, _ctx| {
+                act.set_key(secret_key);
```

Comment:
Why not use `ctx.notify` as before?

Reply:
When using notify, the actor can receive other messages before that notify, which leads to handling external messages in an uninitialized state. By calling `act.set_key()`, we guarantee a synchronous initialization.